src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -114,11 +125,25 @@ private void Initialize(Compilation comp, AnalyzerExecutor analyzerExecutor, Can
             }
         }
 
-        private Task ExecuteSyntaxTreeActions(CancellationToken cancellationToken)
+        public void StartCompleteAnalysis(ImmutableHashSet<SyntaxTree> syntaxTreesToSkip, CancellationToken cancellationToken)
+        {
+            if (_initializeTaskStarted && _primaryTask == null)
+            {
+                _primaryTask = Task.Run(async () =>
+                    {
+                        await _initializeTask.ConfigureAwait(false);
+                        await ProcessCompilationEventsAsync(true, cancellationToken).ConfigureAwait(false);
+                        await ExecuteSyntaxTreeActions(_compilation.SyntaxTrees.Where((tree) => !syntaxTreesToSkip.Contains(tree)), cancellationToken).ConfigureAwait(false);
```

Comment:
I think this change set is post merge. I removed the call to ExecuteSyntaxTreeActions from ProcessCompilationEventsAsync, but you are correct that what I've got here violates the action execution order rules. So would executing the syntax tree actions before processing any compilation events. So, you are correct that this is not correct.